assessment/collectors/infoschema_collector.go

```diff
@@ -113,26 +131,104 @@ func getInfoSchema(sourceProfile profiles.SourceProfile) (common.InfoSchema, err
 	}
 }
 
-func (c InfoSchemaCollector) ListTables() (map[string]utils.TableDetails, map[string]utils.TableDetails) {
-	srcTable := make(map[string]utils.TableDetails)
-	spTable := make(map[string]utils.TableDetails)
+func (c InfoSchemaCollector) ListTables() (map[string]utils.SrcTableDetails, map[string]utils.SpTableDetails) {
+	srcTable := make(map[string]utils.SrcTableDetails)
+	spTable := make(map[string]utils.SpTableDetails)
 
 	for tableId := range c.tables {
-		srcTable[tableId] = utils.TableDetails{
-			Id:   tableId,
-			Name: c.conv.SrcSchema[tableId].Name,
+		srcCheckConstraints := make(map[string]schema.CheckConstraint)
+		for _, ck := range c.conv.SrcSchema[tableId].CheckConstraints {
+			srcCheckConstraints[ck.Id] = ck
 		}
-		spTable[tableId] = utils.TableDetails{
-			Id:   tableId,
-			Name: c.conv.SpSchema[tableId].Name,
+		spCheckConstraints := make(map[string]ddl.CheckConstraint)
+		for _, ck := range c.conv.SpSchema[tableId].CheckConstraints {
+			spCheckConstraints[ck.Id] = ck
+		}
+		srcFks := make(map[string]utils.SourceForeignKey)
+		for _, fk := range c.conv.SrcSchema[tableId].ForeignKeys {
+			srcFks[fk.Id] = utils.SourceForeignKey{
+				Definition: fk,
+				// TODO : Move all print methods to source specific classes
+				Ddl: utils.PrintForeignKeyAlterTable(fk, tableId, c.conv.SrcSchema),
+			}
+		}
+		spFks := make(map[string]utils.SpannerForeignKey)
+		for _, fk := range c.conv.SpSchema[tableId].ForeignKeys {
+			spFks[fk.Id] = utils.SpannerForeignKey{
+				Definition:      fk,
+				IsInterleavable: canInterleaveWithFK(c.conv.SpSchema, tableId, fk.ReferTableId, &fk),
+				ParentTableName: c.conv.SpSchema[fk.ReferTableId].Name,
+				Ddl:             fk.PrintForeignKeyAlterTable(c.conv.SpSchema, ddl.Config{}, tableId),
+			}
+		}
+		srcTable[tableId] = utils.SrcTableDetails{
+			Id:               tableId,
+			Name:             c.conv.SrcSchema[tableId].Name,
+			Charset:          c.tables[tableId].Charset,
+			Collation:        c.tables[tableId].Collation,
+			CheckConstraints: srcCheckConstraints,
+			SourceForeignKey: srcFks,
+		}
+		spTable[tableId] = utils.SpTableDetails{
+			Id:                tableId,
+			Name:              c.conv.SpSchema[tableId].Name,
+			CheckConstraints:  spCheckConstraints,
+			SpannerForeignKey: spFks,
 		}
 	}
 	return srcTable, spTable
 }
 
+func canInterleaveWithFK(tableSchemaMap map[string]ddl.CreateTable, childTableId string, parentTableId string, fk *ddl.Foreignkey) bool {
```

Comment:
We need to work with what is present in the Spanner schema. So structurally, the collector should just return the definition of the interleaved table.

During the comparison of an interleaved table to an FK - this method would be required - i.e. in the assessment engine.

Reply:
Can be taken as a follow up

Reply:
Added a todo